File: c#2010/PDFFormFields/FormCheckBox.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace WindowsApplication1
{
    public partial class FormCheckBox : Form
    {
        public FormCheckBox()
        {
            InitializeComponent();
        }

        public int Checked
        {
            get
            {
                if (checkBox1.Checked)
                    return 1;
                else
                    return 0;
            }
            set
            {
                if (value == 1)
                {
                  
                    checkBox1.Checked = true;
                }
                else
                {
                 
                    checkBox1.Checked = false;


                }
            }
        }

        public int FieldIndex
        {
            set { lblindex.Text = value.ToString(); }
        }

        private void FormCheckBox_Load(object sender, EventArgs e)
        {

        }

        private void button1_Click(object sender, EventArgs e)
        {
            this.DialogResult = DialogResult.OK;
            this.Close();
        }
    }
}

File: VBdotNet2010/PDFFormFields/FormCheckBox.vb
Public Class FormCheckBox

    Public Property Checked() As Integer
        Get
            If checkBox1.Checked Then
                Return 1
            Else
                Return 0
            End If

        End Get

        Set(ByVal value As Integer)

            If value = 1 Then
                checkBox1.Checked = True
            Else
                checkBox1.Checked = False

            End If
        End Set

    End Property

    Public Property FieldIndex() As Integer
        Get
            Return Val(lblindex.Text)
        End Get

        Set(ByVal value As Integer)

            lblindex.Text = value.ToString()
        End Set

    End Property
    Private Sub button1_Click(sender As System.Object, e As System.EventArgs) Handles button1.Click
        DialogResult = DialogResult.OK
        Close()
    End Sub

    Private Sub FormCheckBox_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load

    End Sub
End Class